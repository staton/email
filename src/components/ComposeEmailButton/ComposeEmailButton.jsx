import React, { Component } from 'react';
import PropTypes from 'prop-types';

const propTypes = {
    content: PropTypes.any.isRequired,
    onClick: PropTypes.func,
    title: PropTypes.string
};

const defaultProps = {
    onClick: () => {},
    title: ''
};

export class ComposeEmailButton extends Component {

    constructor(props) {
        super(props);
    }

	render() {
		return (
            <div className="ComposeEmailButton">
                <button 
                    className="ComposeEmailButton__button"
                    onClick={this.props.onClick}
                    title={this.props.title}
                >
                    {this.props.content}
                </button>
            </div>
		);
    }

}

ComposeEmailButton.propTypes = propTypes;
ComposeEmailButton.defaultProps = defaultProps;

export default ComposeEmailButton;