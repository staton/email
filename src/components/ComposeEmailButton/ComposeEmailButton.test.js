import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import { ComposeEmailButton } from './ComposeEmailButton';

describe('<ComposeEmailButton />', () => {

    it('renders correctly', () => {
        const wrapper = shallow(
            <ComposeEmailButton 
                content="New Email"
                title="Search"
            />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    
    it('handles click events', () => {
        const onClick = jest.fn();
        const wrapper = mount(
            <ComposeEmailButton 
                content="New Email"
                onClick={onClick}
            />);
        let button = wrapper.find('button');

        button.simulate('click');
        expect(onClick).toHaveBeenCalledTimes(1);
    });

});