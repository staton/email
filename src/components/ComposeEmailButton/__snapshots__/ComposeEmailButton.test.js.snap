// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComposeEmailButton /> renders correctly 1`] = `
<div
  className="ComposeEmailButton"
>
  <button
    className="ComposeEmailButton__button"
    onClick={[Function]}
    title="Search"
  >
    New Email
  </button>
</div>
`;
